ddtrace/llmobs/_integrations/pydantic_ai.py

```diff
@@ -145,6 +129,58 @@ def _llmobs_set_tags_tool(
         if not span.error:
             span._set_ctx_item(OUTPUT_VALUE, getattr(response, "content", ""))
 
+    def _tag_agent_manifest(self, span: Span, kwargs: Dict[str, Any], agent: Any) -> None:
+        if not agent:
+            return
+
+        manifest = {}
+        manifest["framework"] = "PydanticAI"
+        if hasattr(agent, "name"):
+            manifest["name"] = agent.name or "PydanticAI Agent"
+        if span.get_tag("pydantic_ai.request.model"):
```

Comment:
not as critical as other scenarios where we've relied on APM span tags in the past, but would it be possible to decouple this logic from APM span tags? is it easy enough to re-get the model and provider metadata?